irohad/multi_sig_transactions/mst_time_provider.hpp

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IROHA_MST_TIME_PROVIDER_HPP
+#define IROHA_MST_TIME_PROVIDER_HPP
+
+#include "multi_sig_transactions/mst_types.hpp"
+
+namespace iroha {
+
+  /**
+   * Interface provides current time for iroha
+   */
+  class MstTimeProvider {
+   public:
+
+    /**
+     * Fetching current time in system
+     * @return current time
+     */
+    virtual TimeType getCurrentTime() const = 0;
```

Comment:
I could not find a definition for this. Also looks suitable to move such method into utils or something like that.